index.js

```diff
@@ -169,21 +163,40 @@ Sequest.prototype.__write = function (chunk, encoding, cb) {
           if (!self.leaveOpen) self.connection.end()
           return self.emit('error', e)
         }
-        stream.on('exit', function (_code, _signal) {
-          signal = _signal
-          code = _code
+        stream.on('exit', function (code, signal) {
           if (code) {
             if (!self.leaveOpen) self.connection.end()
             return self.emit('error', new Error('Exit code non-zero, '+code))
           }
         })
-        stream.on('end', function () {
+        onEndAndExit(stream, function(code, signal) {
           self.emit('exec', e, cmd, code, signal)
           if (!code) cb()
         })
       }
     })
   }
+  function onEndAndExit(stream, cb) {
+    var code
+      , signal
+      , ended = false
+      , exited = false
+      ;
+
+    stream.once('exit', function(_code, _signal) {
+      exited = true
+
+      if (ended) return cb(_code, _signal)
+
+      code = _code
+      signal = _signal
```

Comment:
Minor nit, but for the sake of symmetry, I'd probably move these assignments above the `if (ended)` line and then have both the `.once('end')` and `.once('exit')` callbacks finish with the exact same line:
```js
 if (ended && exited) return cb(code, signal)
```
Means reader doesn't need to read two very similar lines to figure out what's different. Also has benefit of succinctly capturing the `if (ended && exited)` logic which is the key piece to understand about this function.